clap_derive/src/attr.rs

```diff
@@ -86,6 +86,7 @@ impl Parse for ClapAttr {
         let magic = match name_str.as_str() {
             "rename_all" => Some(MagicAttrName::RenameAll),
             "rename_all_env" => Some(MagicAttrName::RenameAllEnv),
+            "required" => Some(MagicAttrName::Required),
```

Comment:
Ideally, we don't hard code every build method but stuff with derive-specific behavior, so what you want instead is to add a `group_methods: Vec<Method>` to `Item` that are pushed to when the `AttrKind::Group` is used.

We then apply these group methods similarly to applying `methods` in other cases